src/System.Device.Gpio/System/Device/Gpio/ExtendedPinMode.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Device.Gpio
+{
+    /// <summary>
+    /// Platform-dependent description of an alternate pin mode (i.e. a pin that is used for I2C or SPI transfer).
+    /// Depending on the hardware in use, the use will reflect the effective mode ("PWM", "SPI") or the state of the
+    /// mode multiplexer of the CPU ("ALT0", "ALT1").
+    /// </summary>
+    public class ExtendedPinMode : IEquatable<ExtendedPinMode>
```

Comment:
I'd suggest abstract

Reply:
I need to create the "Unknown" instance below, therefore this is not abstract. Since the ctor is protected, the effect is the same.

Reply:
There is no problem with having static readonly members on abstract class

Reply:
Sure, the problem is not having the static readonly member, but constructing its instance. The `UnknownExtendedPinMode` class would work around this, but I don't really see the benefit of an additional class just to make this abstract.